lambda_function.py

```diff
@@ -64,12 +64,68 @@ def handler(event, context):
         if "ip" in message and not message["ip"]:
             message.pop("ip")
         message_date = str(datetime.fromisoformat(message_time).date())
-        index = index_prefix + message_date
+        index = es_index_prefix + message_date
         action = {"_index": index, "_id": id, "_source": message}
         actions.append(action)
 
     success, errors = helpers.bulk(client, actions, max_retries=3, raise_on_error=False)
     if errors:
         logger.error(errors)
     total = len(event["Records"])
-    print(f"Success processed {success}/{total} items.")
+    print(f"Successfully processed {success}/{total} items for opensearch")
+
+
+def splunk_handler(event, context):
+    secret = get_secret('splunk-quayio-audit-log')
+    # splunk HEC configuration
+    splunk_hec_url = secret["splunk_hec_url"]
+    splunk_hec_token = secret["splunk_hec_token"]
+    splunk_index = secret["splunk_index"]
+    events = []
+
+    for record in event["Records"]:
+        try:
+            # Kinesis data is base64-encoded, so decode here
+            message = json.loads(base64.b64decode(record["kinesis"]["data"]))
+            id = message["random_id"]
+            message_time = message["datetime"]
+            message["@timestamp"] = message_time
+            if "ip" in message and not message["ip"]:
+                message.pop("ip")
+            message_date = str(datetime.fromisoformat(message_time).date())
```

Comment:
This logic duplicates a block in `elastisearch_handler` so this could be refactored to be a function

Reply:
agreed, moved it to a helper func()